src/modules/Game.class.js

```diff
@@ -20,26 +26,358 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(initialState, winCallback, loseCallback) {
+    this.board = initialState || [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.score = 0;
+
+    this.status = IDLE;
+    this.isStarted = false;
+    this.winCallback = winCallback;
+    this.loseCallback = loseCallback;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveTile(fromCell, toCell) {
+    // Отримуємо координати клітинок
+    const fromCellRect = fromCell.getBoundingClientRect();
+    const toCellRect = toCell.getBoundingClientRect();
 
-  /**
-   * @returns {number}
-   */
-  getScore() {}
+    // Вираховуємо зміщення
+    const xOffset = toCellRect.left - fromCellRect.left;
+    const yOffset = toCellRect.top - fromCellRect.top;
+
+    // Встановлюємо зміщення через CSS перемінні
+    fromCell.style.setProperty('--x-offset', `${xOffset}px`);
+    fromCell.style.setProperty('--y-offset', `${yOffset}px`);
+
+    // Додаємо клас для запуску анімації
+    fromCell.classList.add('moving');
+
+    // Затримка для застосування анімації збільшення
+    setTimeout(() => {
+      // Додаємо клас 'merged' для анімації збільшення
+      toCell.classList.add('merged');
+
+      // Затримка для видалення класу 'merged' після завершення анімації
+      setTimeout(() => {
+        toCell.classList.remove('merged');
+      }, 300); // Час анімації (300ms)
+    }, 120); // Затримка до початку анімації
+  }
+
+  updateBoard() {
+    const cells = document.querySelectorAll('.field-cell');
+    // всі клітинки дошки
+
+    let cellIndex = 0;
+
+    for (let row = 0; row < this.board.length; row++) {
+      for (let col = 0; col < this.board[row].length; col++) {
+        const cell = cells[cellIndex++];
+        const value = this.board[row][col];
+        const previousValue = cell.textContent;
+
+        cell.classList.remove(`field-cell--${previousValue}`);
+
+        if (value === 0) {
+          cell.textContent = '';
+        } else {
+          cell.classList.add(`field-cell--${value}`);
+          cell.textContent = value; // оновлюємо текст
```

Comment:
it's clear without comment 🙂 And it's related to 90% of comments - no need to comment smth if it's clear from the code. If there is complicated logic, unusual behavior, etc. we need to add comments

Reply:
oh, sorry, I wrote it for myself and forgot to delete it